tests/acceptance/test_examples.py

```diff
@@ -0,0 +1,60 @@
+"""Acceptance tests for shipping examples."""
+
+import re
+import runpy
+from pathlib import Path
+
+import pytest
+
+import nidaqmx
+
+EXAMPLES_DIRECTORY = Path(__file__).parent.parent.parent / "nidaqmx_examples"
+EXAMPLE_PATHS = [p for p in EXAMPLES_DIRECTORY.glob("**/*.py") if p.name != "__init__.py"]
+
+
+@pytest.mark.parametrize("example_path", EXAMPLE_PATHS)
+def test__shipping_example__run__no_errors(example_path: Path):
+    """Test that a shipping example does not error."""
+    local_system = nidaqmx.system.System.local()
+    example_source = example_path.read_text()
+    for device_name in _find_device_names(example_source):
+        if device_name not in local_system.devices:
+            pytest.skip(f"Cannot find device {device_name}.")
+        device = local_system.devices[device_name]
+        for physical_channel_name in _find_physical_channel_names(example_source, device_name):
+            if not _has_physical_channel(device, physical_channel_name):
+                pytest.skip(
+                    f"Device {device_name} does not have physical channel {physical_channel_name}"
+                )
+    if example_path.name == "ci_pulse_freq.py":
+        pytest.skip("Example times out if there is no signal.")
+    if example_path.name == "every_n_samples_event.py":
+        pytest.skip("Example waits for keyboard input.")
+
+    runpy.run_path(example_path)
+
+
+def _find_device_names(source: str) -> set[str]:
+    return set(re.findall(r"cDAQ\d+Mod\d+|Dev\d+|PXI\d+Slot\d+|TS\d+Mod\d+", source))
+
+
+def _find_physical_channel_names(source: str, device_name: str) -> set[str]:
+    return set(
+        re.findall(
+            device_name + r"/(?:ai\d+|ao\d+|ctr\d+|port\d+(?:/line\d+)?|power)(?::\d+)?", source
+        )
+    )
+
+
+def _has_physical_channel(device: nidaqmx.system.Device, physical_channel_name: str) -> bool:
+    collections = [
+        device.ai_physical_chans,
+        device.ao_physical_chans,
+        device.ci_physical_chans,
+        device.co_physical_chans,
+        device.di_ports,
+        device.di_lines,
+        device.do_ports,
+        device.do_lines,
+    ]
```

Comment:
@zhindes @maxxboehme How do you check which power channels are supported on a device?

Reply:
There is only one device with 1 power channel called "power". If we add a device with multiple channels, we'll add a `power_physical_chans` but didn't bother for now. EDIT: It might show up in `ai_physical_chans` now that I think about it. Under the hood Power is AI. [`sim_ts_power_device`](https://github.com/ni/nidaqmx-python/blob/master/tests/conftest.py#L95.) uses `ai_meas_types` to find a device/

Reply:
It doesn't show up in `ai_physical_chans`. I added a workaround that uses `ai_meas_types`.